ibis/backends/clickhouse/__init__.py

```diff
@@ -54,6 +127,52 @@ def __init__(self, *args, external_tables=None, **kwargs):
     def _register_in_memory_table(self, table_op):
         self._external_tables[table_op.name] = table_op.data.to_frame()
 
+    def _log(self, sql: str) -> None:
+        """Log the SQL, usually to the standard output.
+
+        This method can be implemented by subclasses. The logging
+        happens when `ibis.options.verbose` is `True`.
+        """
+        util.log(sql)
+
+    def sql(self, query: str, schema=None) -> ir.Table:
+        if schema is None:
+            schema = self._get_schema_using_query(query)
+        return ops.SQLQueryResult(query, ibis.schema(schema), self).to_expr()
+
+    def _from_url(self, url: str) -> BaseBackend:
+        """Connect to a backend using a URL `url`.
+
+        Parameters
+        ----------
+        url
+            URL with which to connect to a backend.
+
+        Returns
+        -------
+        BaseBackend
+            A backend instance
+        """
+        url = sa.engine.make_url(url)
+
+        kwargs = {
+            name: value
+            for name in ("host", "port", "database", "password")
+            if (value := getattr(url, name, None))
+        }
+        if username := url.username:
+            kwargs["user"] = username
+
+        kwargs.update(url.query)
+        self._convert_kwargs(kwargs)
+        return self.connect(**kwargs)
+
+    def _convert_kwargs(self, kwargs):
```

Comment:
Is it used anywhere else than `_from_url`?

Reply:
No

Reply:
It's also not a new API in this PR, so I'm inclined to leave it as is.